cmd/nodepool/core/create.go

```diff
@@ -142,3 +144,29 @@ func (o *CreateNodePoolOptions) CreateNodePool(ctx context.Context, platformOpts
 	fmt.Printf("NodePool %s created\n", o.Name)
 	return nil
 }
+
+// ValidateMgmtClusterSupportsNodePoolCPUType validates the HostedCluster payload type can support the CPU architecture
+// of the NodePool.
+func ValidateMgmtClusterSupportsNodePoolCPUType(ctx context.Context, name, namespace, arch string) error {
+	logger := ctrl.LoggerFrom(ctx)
+
+	client, err := util.GetClient()
+	if err != nil {
+		return err
+	}
+
+	hc := &hyperv1.HostedCluster{}
+	err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, hc)
+	if err != nil {
+		// This is expected to happen when we create a cluster since there is no created HostedCluster CR to check the
+		// payload from.
+		logger.Info("WARNING: failed to get HostedCluster to check payload type")
+		return nil
+	}
+
+	if hc.Status.PayloadArch != "" && !strings.EqualFold(hc.Status.PayloadArch, "multi") && !strings.EqualFold(hc.Status.PayloadArch, arch) {
```

Comment:
For the second part of this if condition, you could just compare against a const in the API, rather than a case insensitive match. Typically you'd create a string type alias for enums and have each valid value as an exported constant so that in code you can compare them.

Likewise where we have conversions from API casing to some other casing, this is normally done with a function that knows the correct conversions, with a switch statement, it doesn't matter too much here but for other places where it's not just a casing change that's what I'd suggest

Reply:
> you could just compare against a const in the API, rather than a case insensitive match.

It would still need to be a case insensitive check. The release payloads list the architectures in lowercase. You previously mentioned the enum in the API should have all the fields capitalized. So there's a discrepancy between those two things, which is why I think it would still need to be a case insensitive check.

Reply:
The enum in the status.payload is not consistent with the one in nodepool.spec.Arch one is using lowercase, other one is using uppercase
Also we already have declared consts in nodepool_types
	ArchitectureAMD64   = "amd64"
	ArchitectureS390X   = "s390x"
	ArchitecturePPC64LE = "ppc64le"
	ArchitectureARM64   = "arm64"

that can be leveraged for comparisons
Can we please also add multi to that list and use it consistently? I can see  several "multi" and "Multi" refs now

Reply:
I mean you would have a function, something along the lines of below, that makes explicit conversions rather than just relying on the assumption that casing is always comparable

```
func apiArchToManifestArch(apiArch apiv1.CPUArch) apiv1.ManifestArchitecture {
  switch apiArch {
  case apiv1.ArchitectureAMD64:
    return apiv1.ManifestArchitectureAMD64
  case apiv1.ArchitectureS390X:
    return apiv1.ManifestArchitectureS390X
  ...
  }
}
```

Reply:
Yes let's do that. We also have a similar ArchAliases for rhcos metadata.

Reply:
Yeah I will add that.

I will change the enums back to lowercase since that is what the OCP release image payload does. That makes the most sense to me.

Reply:
I think I got this all squared away in a new commit - [Add ArchitectureMulti to NodePool API](https://github.com/openshift/hypershift/pull/4538/commits/286e3e28e1bbee9a5f91e53f4c1c4083e985c21c). 

Please let me know if I missed something or misunderstood something 😄